virtual/sound-server/sound-server-0.ebuild

```diff
@@ -0,0 +1,17 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+DESCRIPTION="Virtual for Sound Server"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~ia64 ~mips ~ppc ~ppc64 ~riscv ~sparc ~x86 ~amd64-linux ~x86-linux ~ppc-macos ~x64-macos ~sparc-solaris"
+IUSE="alsa bluetooth"
+
+RDEPEND="
+	|| (
+		media-video/pipewire[alsa(+)?,bluetooth?,sound-server(+)]
```

Comment:
I do not think it's a good idea to expect alsa IUSE on pipewire - the fundamental issue is that even if you wanted one sound daemon to have Bluetooth and the other ALSA (I can't think of any realistic reason but suppose you wanted to do that), you'd still be left with the fact that the most common and overall reliable Linux audio API is PulseAudio for which there can be only one provider: either /usr/bin/pulseaudio or pipewire-pulse.

For that reason I always suggest thinking of either running all audio via PipeWire or all through PulseAudio.

Yes, technically one can do exotic setups where pulseaudio and pipewire-pulse co-exist but it requires getting environment variables and socket paths always correct, so it's mostly a "look at what weird thing I can make [by understanding how the audio stack works]" type of thing which doesn't have any good usecase beyond clickbait. And even then, as I know you're aware, WirePlumber and /usr/bin/pulseaudio really do not like to share ALSA access (and they, I think, absolutely can't share BlueZ access).

Reply:
Sorry I do not follow, this construct is just explicitly asking for `pipewire` with USE `alsa` if that exists. Indeed, `pipewire` lacks IUSE `alsa` for some reason and instead enables alsa unconditionally.

Reply:
Yes, as I have explained in multiple places many times to many Gentoo developers/contributors, the reason for that is because we'd likely need REQUIRED_USE to get dependencies right, if there was an alsa IUSE, additionally there could no longer be pipewire-alsa which, if I remember correctly, only exists because I could sneak it in as a build system change when it's really not pipewire-alsa but a related code with the ALSA plugin merely coming along for the ride.

Removal of pipewire-pulse IUSE would no longer be a problem for PipeWire users, since the plugin is now in overall good state. However it would make it even harder for PipeWire and PulseAudio to co-exist, since then PW would always install its ALSA plugin, which you yourself hooked up to be always enabled default.